Content/Items/Magnet/Weapons.GrayGoo.cs

```diff
@@ -0,0 +1,435 @@
+					targetCenter = NPC.Center;
+				}
+				else
+				{
+					EnemyWhoAmI = -1;
+					foundTarget = false;
+				}
+			}
+
+			else if (!Owner.HasMinionAttackTargetNPC)
+			{
+				NPC target = Main.npc.Where(n => n.CanBeChasedBy() && n.Distance(Owner.Center) < maxMinionChaseRange && Collision.CanHitLine(Projectile.position, 0, 0, n.position, 0, 0) && !alreadyTargetted.Contains(n)).OrderBy(n => Vector2.Distance(n.Center, Projectile.Center)).FirstOrDefault();
+				if (target != default)
+				{
+					targetCenter = target.Center;
+					EnemyWhoAmI = target.whoAmI;
+					foundTarget = true;
+				}
+				else
+				{
+					EnemyWhoAmI = 0;
+					foundTarget = false;
+				}
+			}
+
+			if (EnemyWhoAmI != oldEnemyWhoAmI)
+			{
+				if (foundTarget)
+					ReadjustDust();
+				oldEnemyWhoAmI = EnemyWhoAmI;
+			}
+
+			#endregion
+
+			if (!Main.dedServ && lerper < 1)
+			{
+				if (lerper == 0)
+					KillDust();
+				for (int i = 0; i < 5; i++)
+				{
+					Vector2 startPos = Projectile.Center + Main.rand.NextVector2Circular(20, 20);
+					Vector2 offset = Main.rand.NextVector2Circular(20, 20);
+					var dust = Dust.NewDustPerfect(startPos, ModContent.DustType<GrayGooDust>(), Vector2.Zero, 0, Color.Transparent, 1);
+					dust.customData = new GrayGooDustData((int)offset.X, (int)offset.Y, Projectile, Main.rand.NextFloat(0.07f, 0.17f), Main.rand.NextFloat(9, 20));
+				}
+
+				lerper += 0.1f;
+			}
+
+			if (foundTarget)
+			{
+				NPC actualTarget = Main.npc[(int)EnemyWhoAmI];
+
+				if (!actualTarget.active)
+				{
+					EnemyWhoAmI = -1;
+					foundTarget = false;
+					Projectile.velocity = Vector2.Zero;
+				}
+
+				Projectile.velocity = Vector2.Lerp(Projectile.velocity, Projectile.DirectionTo(actualTarget.Center) * 16, 0.07f);
+			}
+			else
+			{
+				Projectile.velocity = Vector2.Lerp(Projectile.velocity, Projectile.DirectionTo(Owner.Center) * 16, 0.02f);
+				EnemyWhoAmI = -1;
+			}
+		}
+
+		public override void OnHitNPC(NPC target, int damage, float knockback, bool crit)
+		{
+			float scale = 1;
+			int amount = 5;
+
+			if (target.life <= 0)
+			{
+				amount = 12;
+				scale = 1.5f;
+			}
+
+			for (int i = 0; i < amount; i++)
+			{
+				Vector2 dir = Main.rand.NextVector2Circular(6, 6);
+				Dust.NewDustPerfect(target.Center + dir, ModContent.DustType<GrayGooSplashDust>(), dir, default, default, scale);
+			}
+		}
+
+		private void ReadjustDust()
+		{
+			NPC target = Main.npc[(int)EnemyWhoAmI];
+			foreach (Dust dust in Main.dust)
+			{
+				if (dust.type == ModContent.DustType<GrayGooDust>() && dust.customData is GrayGooDustData data && data.proj == Projectile)
+				{
+					Vector2 offset = Main.rand.NextVector2Circular(target.width / 2, target.height / 2);
+					data.x = (int)offset.X;
+					data.y = (int)offset.Y;
+				}
+			}
+		}
+
+		private void KillDust()
+		{
+			foreach (Dust dust in Main.dust)
+			{
+				if (dust.type == ModContent.DustType<GrayGooDust>() && dust.customData is GrayGooDustData data && data.proj == Projectile)
```

Comment:
single line conditional should be inlined

Reply:
fixed